src/carousel/carousel.ts

```diff
@@ -196,6 +212,48 @@ export class NgbCarousel implements AfterContentChecked,
     return isFirstSlide ? (this.wrap ? slideArr[slideArr.length - 1].id : slideArr[0].id) :
                           slideArr[currentSlideIdx - 1].id;
   }
+
+  private _getSlideEventDirection(currentActiveSlideId: string, nextActiveSlideId: string):
+      NgbSlideCarouselEventDirection {
+    const currentActiveSlideIdx = this._getSlideIdxById(currentActiveSlideId);
+    const nextActiveSlideIdx = this._getSlideIdxById(nextActiveSlideId);
+
+    return currentActiveSlideIdx > nextActiveSlideIdx ? NgbSlideCarouselEventDirection.RIGHT :
+                                                        NgbSlideCarouselEventDirection.LEFT;
+  }
+}
+
+/**
+* The payload of the slide event fired when the slide transition is completed
+*/
+export interface NgbSlideCarouselEvent {
+  /**
+   * Event type
+   */
+  type: 'slide';
+
+  /**
+   * Previous slide id
+   */
+  prevSlideId: string;
+
+  /**
+   * New slide ids
+   */
+  newSlideId: string;
+
+  /**
+   * Slide event direction
+   */
+  direction: NgbSlideCarouselEventDirection;
+}
+
+/**
+ * Enum to define the carousel slide event direction
+ */
+export enum NgbSlideCarouselEventDirection {
+  LEFT = <any>'left',
+  RIGHT = <any>'right'
```

Comment:
but doing like this NgbSlideCarouselEventDirection.left and NgbSlideCarouselEventDirection.right values won't be 0 and 1 instead of 'left' and 'right'?